Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -282,13 +281,38 @@ struct ReactantProductAtomMapping {
   std::map<unsigned int, std::vector<unsigned int> > reactProdAtomMap;
   std::map<unsigned int, unsigned int> prodReactAtomMap;
   std::map<unsigned int, unsigned int> prodAtomBondMap;
+
+  // maps (atom map number,atom map number) pairs in the reactant template
+  // to whether or not they are bonded in the template.
+  std::map<std::pair<unsigned int, unsigned int>, unsigned int>
+      reactantTemplateAtomBonds;
 };
 
 ReactantProductAtomMapping *getAtomMappingsReactantProduct(
     const MatchVectType &match, const ROMol &reactantTemplate,
     RWMOL_SPTR product, unsigned numReactAtoms) {
   ReactantProductAtomMapping *mapping =
       new ReactantProductAtomMapping(numReactAtoms);
+
+  // keep track of which mapped atoms in the reactant template are bonded to
+  // each other.
+  // This is part of the fix for #1387
+  {
+    ROMol::EDGE_ITER firstB, lastB;
+    boost::tie(firstB, lastB) = reactantTemplate.getEdges();
+    while (firstB != lastB) {
+      BOND_SPTR bond = reactantTemplate[*firstB];
+      // this will put in pairs with 0s for things that aren't mapped, but we
+      // don't care about that
+      int a1mapidx = bond->getBeginAtom()->getAtomMapNum();
+      int a2mapidx = bond->getEndAtom()->getAtomMapNum();
+      if (a1mapidx > a2mapidx) std::swap(a1mapidx, a2mapidx);
```

Comment:
since this will always be checked, I would not add to the map if a1mapidx or a2mapidx are 0.